lib/ui/pedals/rings.lua

```diff
@@ -0,0 +1,91 @@
+--- RingsPedal
+-- @classmod RingsPedal
+
+local ControlSpec = require "controlspec"
+local UI = require "ui"
+local Pedal = include("lib/ui/pedals/pedal")
+local Controlspecs = include("lib/ui/util/controlspecs")
+
+local RingsPedal = Pedal:new()
+-- Must match this pedal's .sc file's *id
+RingsPedal.id = "rings"
+
+function RingsPedal:new(bypass_by_default)
+  local i = Pedal:new(bypass_by_default)
+  setmetatable(i, self)
+  self.__index = self
+
+  i.sections = {
+    {"Pitch/Struct", "Bright/Damp"}, 
+    {"Position", "Model/EasterEgg"},
+    i:_default_section(),
+  }
+  i:_complete_initialization()
+
+  return i
+end
+
+function RingsPedal:name(short)
+  return short and "RNGS" or "Rings"
+end
+
+function RingsPedal.params()
+  local id_prefix = RingsPedal.id
+
+  local pitch_control = {
+    id = id_prefix .. "_pit",
+    name = "Pitch",
+    type = "control",
+    controlspec = ControlSpec.new(0, 127, "lin", 1, 60, ""),
```

Comment:
in my experience, any control with a range larger than 100 possible values doesn't play nice with norns' encoders (some values are skipped). Elsewhere in the app I've narrowed the range to 24-103 🤷 
```suggestion
    controlspec = ControlSpec.new(24, 103, "lin", 1, 60, ""), -- c3 by default
```

Reply:
controlspec.MIDINOTE ?